depends/packages/solc.mk

```diff
@@ -0,0 +1,37 @@
+package=solc
+$(package)_version=0.8.20
+$(package)_download_path=https://github.com/ethereum/solidity/releases/download/v$($(package)_version)/
+
+# NOTE: solc gets invoked on the BUILD OS during compile. So, we don't
+# care about HOST OS, unlike all other dependencies.
+# That is: solc is run on the BUILD OS, not targeted to run on the
+# HOST OS.
+
+ifeq ($(build_os)-$(build_arch),linux-x86_64)
+  $(package)_file_name=solc-static-linux
+  $(package)_sha256_hash=0479d44fdf9c501c25337fdc540419f1593b884a87b47f023da4f1c700fda782
+endif
+ifeq ($(build_os),darwin)
```

Comment:
```suggestion

# Note: solc only provides binaries for darwin amd64, however since rosetta is
# expected to be present, this should work through emulation without additional config.
ifeq ($(build_os),darwin)
```